server/testsupport/curiousAssessmentsTestDataBuilder.ts

```diff
@@ -0,0 +1,469 @@
+    assessmentDate: '2012-02-16',
+  },
+): LearnerAssessmentV1DTO => ({
+  establishmentId: options.prisonId,
+  establishmentName: options.prisonName,
+  qualification: {
+    qualificationType: options.qualificationType,
+    qualificationGrade: options.qualificationGrade,
+    assessmentDate: options.assessmentDate,
+  },
+})
+
+const aLearnerLddInfoExternalV1DTO = (
+  options: {
+    prisonId?: string
+    prisonName?: string
+    lddPrimaryName?: string
+    lddSecondaryNames?: string[]
+    inDepthAssessmentDate?: string
+    rapidAssessmentDate?: string
+  } = {
+    prisonId: 'MDI',
+    prisonName: 'MOORLAND (HMP & YOI)',
+    lddPrimaryName: 'Visual impairment',
+    lddSecondaryNames: ['Hearing impairment', 'Mental health difficulty', 'Social and emotional difficulties'],
+    inDepthAssessmentDate: '2012-02-16',
+    rapidAssessmentDate: '2012-02-16',
+  },
+): LearnerLddInfoExternalV1DTO => ({
+  establishmentId: options.prisonId,
+  establishmentName: options.prisonName,
+  lddPrimaryName: options.lddPrimaryName,
+  lddSecondaryNames: options.lddSecondaryNames,
+  inDepthAssessmentDate: options.inDepthAssessmentDate,
+  rapidAssessmentDate: options.rapidAssessmentDate,
+})
+
+const aLearnerAssessmentV2DTO = (
+  options: {
+    prisonNumber?: string
+    assessments?: ExternalAssessmentsDTO
+  } = {
+    prisonNumber: 'A1234BC',
+    assessments: anExternalAssessmentsDTO(),
+  },
+): LearnerAssessmentV2DTO => ({
+  prn: options.prisonNumber,
+  assessments: options.assessments,
+})
+
+const anExternalAssessmentsDTO = (
+  options: {
+    alnAssessments?: Array<LearnerAssessmentsAlnDTO>
+    digitalFunctionalSkillsAssessments?: Array<LearnerAssessmentsFunctionalSkillsDTO>
+    englishFunctionalSkills?: Array<LearnerAssessmentsFunctionalSkillsDTO>
+    mathsFunctionalSkills?: Array<LearnerAssessmentsFunctionalSkillsDTO>
+    esolAssessments?: Array<LearnerAssessmentsDTO>
+    readingAssessments?: Array<LearnerAssessmentsDTO>
+  } = {
+    alnAssessments: [anAlnLearnerAssessmentsDTO()],
+    digitalFunctionalSkillsAssessments: [aDigitalFunctionalSkillsLearnerAssessmentsDTO()],
+    englishFunctionalSkills: [anEnglishFunctionalSkillsLearnerAssessmentsDTO()],
+    mathsFunctionalSkills: [aMathsFunctionalSkillsLearnerAssessmentsDTO()],
+    esolAssessments: [anEsolLearnerAssessmentsDTO()],
+    readingAssessments: [aReadingLearnerAssessmentsDTO()],
+  },
+): ExternalAssessmentsDTO => ({
+  aln: options.alnAssessments,
+  digitalSkillsFunctionalSkills: options.digitalFunctionalSkillsAssessments,
+  englishFunctionalSkills: options.englishFunctionalSkills,
+  mathsFunctionalSkills: options.mathsFunctionalSkills,
+  esol: options.esolAssessments,
+  reading: options.readingAssessments,
+})
+
+const anAlnLearnerAssessmentsDTO = (
+  options: {
+    prisonId?: string
+    prisonName?: string
+    assessmentDate?: string
+    assessmentOutcome?: 'Yes' | 'No'
+    hasPrisonerConsent?: 'Yes' | 'No'
+    stakeholderReferral?:
```

Comment:
Here's an example of a subtle difference in field values across the types.
All assessment types have the field `stakeholderReferral`. For most of them they all share the same set of 7 options.
An ALN variant though (`LearnerAssessmentsAlnDTO`) has an 8th option!

Also, the Digital Functional Skills one, even though the type has the field, it is alway null ! 🙄